src/libs/ajax/Catalog.ts

```diff
@@ -0,0 +1,169 @@
+import * as _ from 'lodash/fp'
+import { authOpts, fetchCatalog, jsonBody } from 'src/libs/ajax/ajax-common'
+
+
+interface DatasetTableListEntry {
```

Comment:
I was thinking about pulling these types out into a separate file. Curious to hear thoughts/preferences there.

Reply:
I am totally fine with types being near the top of the main file that uses them.... though they should almost always be exported for use in the rest of the code-base.

Reply:
This seems fine to me for now, although I can see in the long term that it would make sense to split it up. The APIs won't change much over time but the schema may change. We may want to add more comments etc alongside the schema too.

Reply:
Sounds good! I can export them